core/src/test/java/apoc/path/ExpandPathTest.java

```diff
@@ -525,4 +525,71 @@ public void testRelationshipFilterWorksWithoutTypeIncoming() {
 					assertEquals(1, maps.size());
 				});
 	}
+
+	@Test
+	public void testLabelWithSpecialChar() {
+		db.executeTransactionally(
+				"CREATE (n:`http://example.com/abc#Object` {one: 'alpha'})-[:REL]->(o:`http://www.w3.org/2002/07/owl#Class`:OwlClass {two: 'beta'})"
```

Comment:
I would write the test setup slightly different so there are some nodes with e.g. the labels 
`http` +` //example.com/abc#Object`, only `http` or other similar cases that should be filtered out by the label filter. That way it would be easier to notice if there are any bugs around this solution.

Reply:
Added